sys/include/net/gnrc/nib.h

```diff
@@ -0,0 +1,181 @@
+                                                                 */
+#define GNRC_NIB_FLAGS_NC_STATE_STALE           (0x00000030U)   /**< The entry's L2 info is stale */
+#define GNRC_NIB_FLAGS_NC_STATE_DELAY           (0x00000040U)   /**< entry is stale, but packet was
+                                                                 *   send out */
+#define GNRC_NIB_FLAGS_NC_STATE_PROBE           (0x00000050U)   /**< NUD is performed */
+#define GNRC_NIB_FLAGS_NC_STATE_REACHABLE       (0x00000070U)   /**< Entry is reachable */
+/** @} */
+
+/**
+ * @brief   This entry is a router
+ *
+ * This flag is different from @ref GNRC_NIB_FLAGS_DEFAULT_ROUTER_LIST in the
+ * sense that @ref GNRC_NIB_FLAGS_IS_ROUTER is used by NUD to determine if an
+ * entry is *still* a router to manipulate forwarding information, in case this
+ * isn't the case anymore. @ref GNRC_NIB_FLAGS_DEFAULT_ROUTER_LIST on the other
+ * hand is used for marking the packet as part of the default router list i.e.
+ * that it is used to make a forwarding decision.
+ */
+#define GNRC_NIB_FLAGS_IS_ROUTER                (0x00000080U)
+
+/**
+ * @brief   6LoWPAN neighbor cache types
+ * @see <a href="http://tools.ietf.org/html/rfc6775#section-3.5">
+ *          RFC 6775, section 3.5
+ *      </a>
+ * @{
+ */
+#define GNRC_NIB_FLAGS_NC_TYPE_MASK             (0x00000300U)
+#define GNRC_NIB_FLAGS_NC_TYPE_NONE             (0x00000000U)
+#define GNRC_NIB_FLAGS_NC_TYPE_GC               (0x00000100U)   /**< The entry is marked for removal
+                                                                 */
+#define GNRC_NIB_FLAGS_NC_TYPE_TENTATIVE        (0x00000200U)   /**< The entry is temporary */
+#define GNRC_NIB_FLAGS_NC_TYPE_REGISTERED       (0x00000300U)   /**< The entry is registered */
+/** @} */
+
+#define GNRC_NIB_FLAGS_PREFIX_VALID             (0x00000400U)   /**< The prefix is valid */
+
+/** @} */
+
+/**
+ * @brief   Link-layer information for a NIB entry
+ */
+typedef struct {
+    uint8_t addr[GNRC_NETIF_HDR_L2ADDR_MAX_LEN];    /**< link-layer address */
+#ifdef MODULE_GNRC_SIXLOWPAN_ND_ROUTER
+    eui64_t eui64;                                  /**< EUI-64 of the node to uniquely identify
+                                                     *   the node */
+#endif
+    size_t addr_len;                                /**< length of gnrc_nib_l2_info_t::addr */
+} gnrc_nib_l2_info_t;
+
+/**
+ * @brief   Event for sending a packet
+ */
+typedef struct {
+    extimer_event_t event;      /**< extimer event */
+    gnrc_pkt_t *pkt;            /**< packet to send */
+} gnrc_nib_send_event_t;
+
+typedef struct {
+    gnrc_ipv6_netif_t *iface;                       /**< the interface associated with this
+                                                     *   information */
+    ipv6_addr_t *prefix;                            /**< prefix or destination */
+    ipv6_addr_t *next_hop;                          /**< next hop to the destination */
```

Comment:
How about using `universal_address_t`s [1] for `prefix` and `next_hop`?
Using this container allows for sharing the stored address with the FIB and other modules.
However, it requires to use the provided functions for thread safety, and creating a new container with an address requires to iterate/search the current container entries.

[1] https://github.com/RIOT-OS/RIOT/blob/master/sys/include/universal_address.h#L63